Makefile

```diff
@@ -119,6 +119,12 @@ ifdef CONFIG_WIN32
 DEFINES+=-D__USE_MINGW_ANSI_STDIO # for standard snprintf behavior
 endif
 
+ifndef CONFIG_WIN32
+ifeq ($(shell $(CC) -o compat/test-closefrom compat/test-closefrom.c 2>/dev/null && echo 1 || echo 0),1)
+DEFINES+=-DHAVE_CLOSEFROM
+endif
+endif
+
```

Comment:
I am more or less OK with this approach but I have some remarks:
- the executable must not be committed to the repository
- it would be better to invoke the compiler without generating an executable: eg: `clang -fsyntax-only` or to generate the executable to `/dev/null` or at least remove the executable after the check.
- it would be less noisy to add a negative definition `-DCLOSEFROM_NOT_AVAILABLE` if `closefrom` is **not** available, which is not the most common case.

Reply:
I agree that generating the binary to /dev/null would be an improvement.
But using a negative definition wouldn't change much since we would still need to handle both usecases

Reply:
Changed the compiler outfile to `/dev/null`.